corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js

```diff
@@ -351,16 +351,20 @@ hqDefine("cloudcare/js/formplayer/menus/views", function () {
             urlObject.addSelection(caseId);
             const fetchingDetails = FormplayerFrontend.getChannel().request("entity:get:details", urlObject, false, true);
             $.when(fetchingDetails).done(function (detailResponse) {
-                self.model.set("data", detailResponse.models[0].attributes.details)
-                console.log('updating case');
+                self.updateModelFromDetailResponse(caseId, detailResponse)
             }).fail(function () {
                 console.log('could not get case details');
             });
         },
 
+        updateModelFromDetailResponse: function (caseId, detailResponse) {
+            this.model.set("data", detailResponse.models[0].attributes.details)
+        },
+
         modelChanged: function() {
-            console.log('rendering case');
-            this.render();
+            if (!this.model.get('updating')) {
```

Comment:
I see `updating` is set in the `CaseTileGroupedView` `updateModelFromDetailResponse`. I'm assuming this flag is needed for some async condition? but would would this/should it retrigger when `updating` changes from true to false within `updateModelFromDetailResponse`?

Reply:
It does trigger `modelChanged` again when setting `updating` to false. And at that point if would actually render. The problem was, that the tile was being rendered with half the update done.

Reply:
ah ok, I see you have this in ` modelEvents:`, I must have missed that before. Thanks for explaining